pkg/secrets-store/nodeserver.go

```diff
@@ -329,3 +273,82 @@ func (ns *nodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstag
 
 	return &csi.NodeUnstageVolumeResponse{}, nil
 }
+
+func (ns *nodeServer) mountSecretsStoreObjectContent(ctx context.Context, providerName, attributes, secrets, targetPath, permission string) (map[string]string, string, error) {
+	if len(attributes) == 0 {
+		return nil, "", errors.New("missing attributes")
+	}
+	if len(targetPath) == 0 {
+		return nil, "", errors.New("missing target path")
+	}
+	if len(permission) == 0 {
+		return nil, "", errors.New("missing file permissions")
+	}
+	// get provider volume path
+	providerVolumePath := ns.providerVolumePath
+	if providerVolumePath == "" {
+		return nil, "", fmt.Errorf("providers volume path not found. Set PROVIDERS_VOLUME_PATH")
+	}
+
+	// if the provider supports and is running grpc server, then communicate with
+	// provider using the grpc client, otherwise fallback to invoking the provider
+	// binary which is how it was initially implemented
+	_, exists := ns.grpcSupportedProviders[providerName]
+	if exists {
+		log.Infof("Using grpc client for provider: %s", providerName)
+		providerClient, err := newProviderClient(csiProviderName(providerName), ns.providerVolumePath)
+		if err != nil {
+			return nil, ProviderError, fmt.Errorf("failed to create provider client, err: %+v", err)
```

Comment:
nit: maybe an error specifically for `ProviderGRPCClientNotFound` similar to `ProviderBinaryNotFound `?